test/NuGet.Core.Tests/NuGet.Packaging.Test/ContentModelTests/ReadOnlyMemoryCharComparerOrdinalTests.cs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using FluentAssertions;
+using NuGet.ContentModel;
+using Xunit;
+
+namespace NuGet.Packaging.Test.ContentModelTests
+{
+    public class ReadOnlyMemoryCharComparerOrdinalTests
+    {
+        [Theory]
+        [InlineData("AbCDeFg", "AbCDeFg", true)]
+        [InlineData("AbCDeFG", "abcdefg", false)]
```

Comment:
I still don't understand how the code works, or why this test passes. Can anybody explain it to me?

The `for` loop processes multiples of 4 characters. Then there's an `if (charactersRemaining > 0)` which runs `num1 = whatever(num1) ^ pSpan0[obj.Length - 1];`

In the case of 7 character test, the `for` loop processes array indexes 0, 1, 2, and 3. `charactersRemaining` should be 3, so, the if statement processes `pSpan0[7 - 1]` or `pSpan0[6]`.

How are characters 4 and 5 contributing to the hash?

Reply:
I'm not sure, I tried debugging it but ran out of time.  I did add a unit test to verify that the hash of `ABCDEFGH` and `ABCDefgH` are not the same which I think shows that its working as expected?